samples/net/socket_echo/src/socket_echo.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#ifndef __ZEPHYR__
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#define init_net()
+
+#else
+
+#include <net/socket.h>
+#include <kernel.h>
+#include <net_sample_app.h>
+
+void init_net(void)
+{
+	int ret = net_sample_app_init("socket_echo", NET_SAMPLE_NEED_IPV4,
+				      K_SECONDS(3));
+
+	if (ret < 0) {
+		printf("Application init failed\n");
+		k_panic();
+	}
+}
+
+#endif
+
+int main(void)
+{
+	int serv;
+	struct sockaddr_in bind_addr;
+
+	init_net();
+
+	serv = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+
+	bind_addr.sin_family = AF_INET;
```

Comment:
I meant having dual IPv4 and IPv6 at the same time. But it requires two sockets at the moment and support for poll/select as you said, so we can postpone this for later stage.